mlir/lib/Dialect/Rock/Transforms/BlockwiseGemmToThreadwise.cpp

```diff
@@ -405,170 +403,124 @@ struct BlockwiseGemmV2RewritePattern
-                                                          b.getListener());
-      auto olniv = outerLoopN.getInductionVar();
-      auto nOffset = olnb.create<AddIOp>(
-          loc, bBase, olnb.create<MulIOp>(loc, nPerMfmaGroupConstantOp, olniv));
-      auto kOffsetB = olnb.create<MulIOp>(loc, olniv, KConstantOp);
-
-      auto innerLoopNK = olnb.create<AffineForOp>(loc, 0, KPerThread);
-      auto ilnkb = ConversionPatternRewriter::atBlockBegin(
-          innerLoopNK.getBody(), olnb.getListener());
-      auto ilnkiv = innerLoopNK.getInductionVar();
-
-      Value sourceOffsetB = ilnkb.create<AddIOp>(
-          loc,
-          ilnkb.create<AddIOp>(
-              loc, ilnkb.create<MulIOp>(loc, ilnkiv, NConstantOp), laneId),
-          nOffset);
-
-      if (KPack > 1)
-        sourceOffsetB = ilnkb.create<MulIOp>(
-            loc, sourceOffsetB, ilnkb.create<ConstantIndexOp>(loc, KPack));
-
-      auto destOffsetB = ilnkb.create<AddIOp>(loc, ilnkiv, kOffsetB);
-
-      Value valueB = ilnkb.create<InBoundsLoadOp>(
-          loc, bufferBElementType, op.getMatrixB(), sourceOffsetB);
-      ilnkb.create<memref::StoreOp>(loc, valueB, bufferB,
-                                    ValueRange{destOffsetB});
-    } else {
-      // const index_t blk_id = laneId / mfma_type.num_threads_blk;
-      // const index_t blk_td = laneId % mfma_type.num_threads_blk;
-      auto inputSpanLenConstantOp =
-          b.create<ConstantIndexOp>(loc, inputSpanLen);
-      auto blk_id = b.create<DivUIOp>(loc, laneId, inputSpanLenConstantOp);
-      auto blk_td = b.create<RemUIOp>(loc, laneId, inputSpanLenConstantOp);
-
-      Value kBaseA = b.create<AddIOp>(loc, aBase, blk_td);
-      Value kBaseB = b.create<AddIOp>(loc, bBase, blk_td);
-
-      // for(index_t k_i = 0; k_i < KPerThread; k_i += mfma_type.num_input_blks)
-      // {
-      //     a[k_i] = p_a_wave[(k_i * num_input_blks + blk_id) * M + blk_td];
-      //     b[k_i] = p_b_wave[(k_i * num_input_blks + blk_id) * N + blk_td];
-      // }
-      // p_a_wave need to be offseted by waveOffsetA.
-      // p_b_wave need to be offseted by waveOffsetB.
-
-      auto inputSpansPerMfmaInConstantOp =
-          b.create<ConstantIndexOp>(loc, inputSpansPerMfmaIn);
-
-      auto loopKLoad = b.create<AffineForOp>(loc, 0, KPerThread);
-      auto lklb = ConversionPatternRewriter::atBlockBegin(loopKLoad.getBody(),
-                                                          b.getListener());
-      auto lkliv = loopKLoad.getInductionVar();
-
-      Value sourceOffsetA = lklb.create<AddIOp>(
-          loc,
-          lklb.create<MulIOp>(
-              loc,
-              lklb.create<AddIOp>(
-                  loc,
-                  lklb.create<MulIOp>(loc, lkliv,
-                                      inputSpansPerMfmaInConstantOp),
-                  blk_id),
-              MConstantOp),
-          kBaseA);
-
+    auto KPerThreadConstantOp = b.create<ConstantIndexOp>(loc, KPerThread);
```

Comment:
Can we get `Value` here, not `auto`? Or do we care? (Not that big a deal)

Reply:
It's not a big deal, but I agree on the principle that auto should be used only when the type is too verbose. I replaced a bunch of `auto` with `Value` so it's more readable .